### PR TITLE
WIP Chat perf 4: for discussion

### PR DESCRIPTION
**Pull request by chrisnojima**
This changes babel so we don't transpile es2015 which is supported natively by electron/node. An example of the speed up is the load of index.bundle.js goes from 2 seconds to 1.5 seconds.

To discuss: Webpack is acting differently. I had to add loader changes to ignore flow and native files and it complains about the dynamic import of the emoji-datasource (which doesn't seem like it should work the old way anyways)

Figured out the weird runtime issues: https://github.com/gaearon/react-hot-loader/issues/435#issuecomment-264636583

@keybase/react-hackers

### CODE REVIEW

File: desktop/package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
     "babel-eslint": "7.1.1",
     "babel-jest": "17.0.2",
     "babel-loader": "6.2.9",
+    "babel-plugin-add-module-exports": "^0.2.1",
```
Comment (chromakode) [inline desktop/package.json]: Where is this used?

File: desktop/.babelrc [hunk 2]
```diff
@@ -1,10 +1,17 @@
 {
-  "presets": ["babel-preset-es2015", "babel-preset-stage-1", "babel-preset-react"],
+  "presets": ["react"],
   "plugins": [
-    "babel-plugin-transform-runtime",
```
Comment (chromakode) [inline desktop/.babelrc]: Why were these removed?

File: desktop/webpack.config.base.js [hunk 3]
```diff
@@ -15,6 +15,12 @@ console.warn('Injecting defines: ', defines)
 module.exports = {
   module: {
     loaders: [{
+      test: /\.flow?$/,
```
Comment (chrisnojima) [inline desktop/webpack.config.base.js]: you mean the resolve extensions? i think we want that still to help with eslint? i'm hesitant to remove it

File: shared/common-adapters/emoji.desktop.js [hunk 4]
```diff
@@ -5,8 +5,40 @@ import {Emoji} from 'emoji-mart'
 
 import type {Props} from 'emoji-mart'
 
-// $FlowIssue with require
-const backgroundImageFn = (set: string, sheetSize: string) => require(`emoji-datasource/sheet_${set}_${sheetSize}.png`)
+type SetType = 'apple' | 'emojione' | 'google' | 'twitter'
+type SheetSizeType = '16' | '20' | '32' | '64'
+const backgroundImageFn = (set: SetType, sheetSize: SheetSizeType) => {
```
Comment (MarcoPolo) [inline shared/common-adapters/emoji.desktop.js]: 👍 thanks

File: desktop/webpack.config.base.js (hunk 3 above)
Comment (chromakode) [inline desktop/webpack.config.base.js]: Ah nvm, I misread the webpack config.

File: shared/common-adapters/emoji.desktop.js [hunk 5]
```diff
@@ -5,8 +5,40 @@ import {Emoji} from 'emoji-mart'
...
+        '64': require('emoji-datasource/sheet_google_64.png'),
+      }[sheetSize]
+    case 'twitter':
+      return {
+        '16': require('emoji-datasource/sheet_twitter_16.png'),
+        '20': require('emoji-datasource/sheet_twitter_20.png'),
+        '32': require('emoji-datasource/sheet_twitter_32.png'),
+        '64': require('emoji-datasource/sheet_twitter_64.png'),
+      }[sheetSize]
+  }
```
Comment (chromakode) [inline shared/common-adapters/emoji.desktop.js]: This block is going to pull in every emoji sheet into our bundle, which I don't think is desired. Can we force our code to only use a particular sheet, and have this function bail if it gets different parameters?

**Comment by chrisnojima** [comment]
this thing causes too many gremlins to try for now. bailing
